api/pagination.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	"math"
+)
+
+type PaginationRequest struct {
+	Page  int    `form:"page"`
+	Limit int    `form:"limit"`
+	Sort  string `form:"sort" validate:"oneof='name ASC' 'name DESC' 'id ASC' 'id DESC' ''"`
```

Comment:
```suggestion
	Sort  string `form:"sort" validate:"oneof='name_ASC' 'name_DESC' 'id_ASC' 'id_DESC' ''"`
```

Not having spaces in the query params is safer for use in a URL, for other two word parameters we use the underscore.

Reply:
I suggest splitting this into multiple fields, one to communicate the sort by field and another to communicate the order. It's useful for scenarios where user wants to sort by more than just name or id, e.g. by created timestamp, without adding more combinations to this field.

```go
type PaginationRequest struct {
    ...
    SortBy  string `form:"sortBy"`
    OrderBy string `from:"orderBy"`
}
```

Reply:
@mxyng What would be the benefit of that vs just having the OrderBy field? I don't think we're going to keep the oneof validation.

Reply:
It removes the need for a formatted input. Values can be validated easily while also being easier for the user to understand and use. The user no longer needs to know which format is supported and we no longer need to worry about spaces or other delimiters